lib/datadog/filter/src/regex.rs

```diff
@@ -0,0 +1,19 @@
+use regex::Regex;
+
+/// Returns compiled word boundary regex.
+pub fn word_regex(to_match: &str) -> Regex {
+    Regex::new(&format!(
+        r#"\b{}\b"#,
+        regex::escape(to_match).replace("\\*", ".*")
+    ))
+    .expect("invalid wildcard regex")
+}
+
+/// Returns compiled wildcard regex.
+pub fn wildcard_regex(to_match: &str) -> Regex {
+    Regex::new(&format!(
+        "^{}$",
+        regex::escape(to_match).replace("\\*", ".*")
+    ))
+    .expect("invalid wildcard regex")
+}
```

Comment:
Hmm, are we not constructing Regex instances at runtime and then dropping them? This appears to be the case and I'd be glad to be wrong. If I am not, what precludes us from knowing what needs matching at boot time? 

It's clear this PR improves throughput but building regex instances, if we could avoid doing it at runtime, would be better still.

Reply:
Thankfully, regexes are only built at boot time.

These helper funcs are only called _before_ returning a `Box<dyn Matcher<T>>`, which is built in a condition's compilation phase (or the equivalent in VRL).

An example is: https://github.com/vectordotdev/vector/blob/36d61da24534bc5af46063c327f4f384b66d7c96/lib/vector-datadog-filter/src/filter.rs#L39-L42

`re` is outside of the hot path; the _string_match_ func returns the matcher block that the compiled regex is moved into:

https://github.com/vectordotdev/vector/blob/36d61da24534bc5af46063c327f4f384b66d7c96/lib/vector-datadog-filter/src/filter.rs#L195-L208